apps/root/src/hooks/useTransactionsHistory.tsx

```diff
@@ -57,7 +59,11 @@ function useTransactionsHistory(): {
 } {
   const transactionService = useTransactionService();
   const [isHookLoading, setIsHookLoading] = React.useState(false);
-  const { isLoading: isLoadingService, history } = transactionService.getStoredTransactionsHistory();
+
+  const { isLoading: isLoadingService, history } = useServiceEvents<TransactionServiceData, TransactionService>(
+    transactionService,
+    'transactionsHistory'
```

Comment:
Talked offline to use the method instead of the property name

Reply:
We are actually returning the useState value and not the class property! I think it's worth keeping it like this, what do you think?

Reply:
internally it is still using the class property so we should change that

Reply:
I've developed a new approach for accessing `servicesData` without directly accessing the property itself. However, I've encountered difficulty in passing a method (such as 'getLabels') as a parameter while maintaining the hook's flexibility for all services. 
This is isolated in the last commit, where I've also removed the useState value since it's no longer utilized if it's not being returned. Let me know if this can be improved! Not sure if this is close to what you had in mind.